src/dev/impl/DevToys/ViewModels/Tools/JsonYaml/JsonYamlToolViewModel.cs

```diff
@@ -0,0 +1,260 @@
+        private async Task TreatQueueAsync()
+        {
+            if (_conversionInProgress)
+            {
+                return;
+            }
+
+            Assumes.NotNull(OutputTextBlock, nameof(OutputTextBlock));
+
+            _conversionInProgress = true;
+
+            await TaskScheduler.Default;
+
+            while (_conversionQueue.TryDequeue(out string text))
+            {
+                bool success;
+                string result;
+                ILanguage language;
+                if (string.Equals(ConversionMode, JsonToYaml, StringComparison.Ordinal))
+                {
+                    success = ConvertJsonToYaml(text, out result);
+                    language = Languages.FindById("yaml");
+                }
+                else
+                {
+                    success = ConvertYamlToJson(text, out result);
+                    language = Languages.JavaScript;
+                }
+
+                ThreadHelper.RunOnUIThreadAsync(ThreadPriority.Low, () =>
+                {
+                    _outputValue = result;
+                    if (success)
+                    {
+                        OutputTextBlock!.SetText(result, language);
+                    }
+                    else
+                    {
+                        OutputTextBlock!.SetText(result);
+                    }
+                }).ForgetSafely();
+            }
+
+            _conversionInProgress = false;
+        }
+
+        private bool ConvertJsonToYaml(string input, out string output)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                output = string.Empty;
+                return false;
+            }
+
+            try
+            {
+                dynamic jsonObject = JsonConvert.DeserializeObject<ExpandoObject>(input);
+                if (jsonObject is not null && jsonObject is not string)
+                {
+                    int indent = 0;
+                    switch (Indentation)
+                    {
+                        case TwoSpaceIndentation:
+                            indent = 2;
+                            break;
+
+                        case FourSpaceIndentation:
+                            indent = 4;
+                            break;
+
+                        default:
+                            throw new NotSupportedException();
+                    }
+
+                    var serializer
+                        = Serializer.FromValueSerializer(
+                            new SerializerBuilder().BuildValueSerializer(),
+                            EmitterSettings.Default.WithBestIndent(indent).WithIndentedSequences());
+
+                    string? yaml = serializer.Serialize(jsonObject);
+                    output = yaml ?? string.Empty;
+                    return true;
+                }
+
+                output = string.Empty;
+            }
+            catch (Exception ex)
+            {
+                output = ex.Message;
```

Comment:
I think it could be better to log this exception (in the app log) and display a user friendly message.

Like this, for example: "Something went wrong during the conversion. For more information you can check the app logs in settings"

We are not sure that the user will know the c# exception message.